### ISSUE TITLE
Addition derived combinators for Monad, Appliicative

### ISSUE THREAD

**Issue by lukewyman** [typelevel/cats#823]
Do we want to have any/all of the following derived combinators for cats structures such as Monad and Applicative?

```
def compose[A,B,C](f: A => F[B], g: B => F[C]): A => F[C]
```
```
def replicateM[A](n: Int, fa: F[A]): F[List[A]]
```
```
def filterM[A](as: List[A])(f: A => F[Boolean]): F[List[A]]
```

**Comment by adelbertc** [comment]
`compose` I think is handled by the `Kleisli` data type

`filterM` we have via `MonadCombine`

`replicateA` would be useful to have, i'm guessing on `Applicative` (maybe `Traverse`)

```scala
def replicateA[F[_] : Applicative, A](n: Int, fa: F[A]): F[List[A]] =
  List.fill(n)(fa).sequence
```

**Comment by lukewyman** [comment]
I took a shot at adding `replicateA`, as noted above, and immediately ran into the following compiler error:

```
/cats/Applicative.scala:35: could not find implicit value for evidence parameter of type cats.Traverse[List]
```
Since `sequence` is dependent on what type of collection the `Traverse` instance is for, it won't work on a concrete type like `List`

Is there another way I should try and implement this, or am I being naive in wanting to add this combinator?

**Comment by ceedubs** [comment]
@lukewyman I think `import cats.std.list._` should make that compiler error go away.

**Comment by lukewyman** [comment]
...and indeed, it did. Thanks, @ceedubs.

**Comment by mikejcurry** [comment]
Should this one be closed now that #830 has been merged - or am I wrong and is there something left on it?

**Comment by ceedubs** [comment]
It sounds like all of these are taken care of except for `filterM`, which is being tracked by #1054, so I'm going to go ahead and close this out. Feel free to reopen if I've missed something.
